tools/toolchains/qnx/qnx_sdp70_common.cmake

```diff
@@ -0,0 +1,60 @@
+set(CMAKE_CXX_COMPILER "${QNX_HOST}/usr/bin/q++")
+set(CMAKE_CXX_COMPILER_TARGET ${arch})
+
+SET(CMAKE_STRIP   "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-strip" CACHE PATH "strip")
+SET(CMAKE_AR      "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-ar" CACHE PATH "archive")
+SET(CMAKE_LINKER  "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-ld" CACHE PATH "linker")
+SET(CMAKE_NM      "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-nm" CACHE PATH "nm")
+SET(CMAKE_OBJCOPY "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-objcopy" CACHE PATH "objcopy")
+SET(CMAKE_OBJDUMP "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-objdump" CACHE PATH "objdump")
+SET(CMAKE_RANLIB  "${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-ranlib" CACHE PATH "ranlib")
+
+SET(CMAKE_C_FLAGS_DEBUG "-g -O0 -fno-builtin")
+SET(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG")
+
+SET(CMAKE_CXX_FLAGS_DEBUG "-g -O0 -fno-builtin")
+SET(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
+
+SET(CMAKE_FIND_ROOT_PATH ${QNX_TARGET})
+SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
+add_compile_options("-D_QNX_SOURCE=1")
+
+# QNX 7 ships with QCC 5.4.0 (based on GCC). Cmake cannot detect the compiler features,
+# so they're added manually. The correct detection can be done if you set as cross-compiler
+# SET(CMAKE_CXX_COMPILER ${QNX_HOST}/usr/bin/nto${CMAKE_SYSTEM_PROCESSOR}-c++)
+# The file with the detected compile features is then in the build directory named CMakeCXXCompiler.cmake
+# A similar bug was reported for Android, see: https://bugreports.qt.io/browse/QTBUG-54666
+
+set(CMAKE_CXX_COMPILE_FEATURES "cxx_std_98;cxx_template_template_parameters;cxx_std_11;cxx_alias_templates;cxx_alignas;cxx_alignof;cxx_attributes;cxx_auto_type;cxx_constexpr;cxx_decltype;cxx_decltype_incomplete_return_types;cxx_default_function_template_args;cxx_defaulted_functions;cxx_defaulted_move_initializers;cxx_delegating_constructors;cxx_deleted_functions;cxx_enum_forward_declarations;cxx_explicit_conversions;cxx_extended_friend_declarations;cxx_extern_templates;cxx_final;cxx_func_identifier;cxx_generalized_initializers;cxx_inheriting_constructors;cxx_inline_namespaces;cxx_lambdas;cxx_local_type_template_args;cxx_long_long_type;cxx_noexcept;cxx_nonstatic_member_init;cxx_nullptr;cxx_override;cxx_range_for;cxx_raw_string_literals;cxx_reference_qualified_functions;cxx_right_angle_brackets;cxx_rvalue_references;cxx_sizeof_member;cxx_static_assert;cxx_strong_enums;cxx_thread_local;cxx_trailing_return_types;cxx_unicode_literals;cxx_uniform_initialization;cxx_unrestricted_unions;cxx_user_literals;cxx_variadic_macros;cxx_variadic_templates;cxx_std_14;cxx_aggregate_default_initializers;cxx_attribute_deprecated;cxx_binary_literals;cxx_contextual_conversions;cxx_decltype_auto;cxx_digit_separators;cxx_generic_lambdas;cxx_lambda_init_captures;cxx_relaxed_constexpr;cxx_return_type_deduction;cxx_variable_templates;cxx_std_17")
```

Comment:
`cxx_std_17` I don't know what this flag exactly causes, but it seems we might omit it.

Reply:
You're probably right.
We don't want "_Compiler mode is at least C++ 17._", yet.